### PR TITLE
Communication - Update IApp to use latest Client interface

### PR DESCRIPTION
**Pull request by aureliusbtc**
**Description**
A clear and concise description of the features you're adding in this pull request.

**Additional context**
Add any other context about the problem you're solving.

**Metadata**
- Fixes #[Link to Issue]

## Summary by CodeRabbit

- **New Features**
  - Introduced new submodules for enhanced smart contract development.
  - Added interfaces and contracts for robust interchain communication, including messaging, execution fees management, and database interactions.
  - Implemented abstract contracts and libraries to support events, modules, and data structures related to interchain transactions.
  - Developed mock contracts and interfaces for testing interchain communication functionalities.

- **Documentation**
  - Configuration settings updated for the Foundry framework to optimize development workflow.

- **Tests**
  - Comprehensive test suites added for interchain communication components, including contracts for messaging, database interactions, execution fees, and module management.
  - Introduced harnesses for detailed testing of library functionalities and interchain communication mechanisms.

- **Chores**
  - Updated package paths and dependencies to ensure smooth integration and development processes.

### CODE REVIEW

File: packages/contracts-communication/contracts/InterchainClientV1.sol [hunk 1]
```diff
@@ -142,15 +142,17 @@ contract InterchainClientV1 is Ownable, IInterchainClientV1 {
...
+        if (srcExecutionService != address(0)) {
+            IExecutionService(srcExecutionService).requestExecution({
+                dstChainId: dstChainId,
+                // TODO: this should be encodedTx.length
+                txPayloadSize: message.length,
+                transactionId: transactionId,
+                executionFee: executionFee,
+                options: options
+            });
+            IExecutionFees(executionFees).addExecutionFee{value: executionFee}(dstChainId, transactionId);
```
Comment (ChiTimesChi) [inline packages/contracts-communication/contracts/InterchainClientV1.sol:154]: There could be a situation now where execution service is not provided, but the execution fee is non-zero. Let's move the addExecutionFee call out of that if statement then?

File: packages/contracts-communication/contracts/InterchainApp.sol [hunk 2]
```diff
@@ -90,8 +92,9 @@ contract InterchainApp {
     event AppMessageSent();
 
     function send(bytes32 receiver, uint256 dstChainId, bytes calldata message) external payable {
+        bytes memory options = OptionsLib.encodeOptionsV1(OptionsV1(200_000, 0));
```
Comment (ChiTimesChi) [inline packages/contracts-communication/contracts/InterchainApp.sol:95]: Small nit:  `OptionsV1(200_000, 0).encodeOptionsV1()` will also work (even without "using OptionsLib" statement)

**Review by ChiTimesChi — APPROVED** [review]
LGTM with a minor change

File: packages/contracts-communication/contracts/InterchainApp.sol (hunk 2 above)
Comment (aureliusbtc) [reply packages/contracts-communication/contracts/InterchainApp.sol:95]: resolved in #2078

**Pull request merged**